app/src/main/res/layout/main_activity.xml

```diff
@@ -121,7 +123,7 @@
             app:layout_constraintBottom_toTopOf="@+id/button"
             app:layout_constraintEnd_toStartOf="@+id/button12"
```

Comment:
Please rename all buttons id's to correct name. "button2" is not good idea for name id.